content/en/blog/2024/getting-started-with-otelsql/index.md

```diff
@@ -0,0 +1,426 @@
+folder. The `client` service will run after the `mysql` service is up and run,
+Then, it will initialize the OpenTelemetry client and otelsql instrumentation,
+make SQL queries to the `mysql` service, and send metrics and trace data to
+`otel-collector` service via
+[OpenTelemetry Protocol (OTLP)](/docs/specs/otel/protocol/).
+
+After receiving the data, the `otel-collector` service will transfer the data
+format and send metrics data to the `prometheus` service, and send trace data to
+the `jaeger` service.
+
+Let's check `main.go` to see what happens in the `client` service. Here is the
+main function.
+
+```go
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer cancel()
+
+	conn, err := initConn(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	shutdownTracerProvider, err := initTracerProvider(ctx, conn)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := shutdownTracerProvider(ctx); err != nil {
+			log.Fatalf("failed to shutdown TracerProvider: %s", err)
+		}
+	}()
+
+	shutdownMeterProvider, err := initMeterProvider(ctx, conn)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := shutdownMeterProvider(ctx); err != nil {
+			log.Fatalf("failed to shutdown MeterProvider: %s", err)
+		}
+	}()
+
+	db := connectDB()
+	defer db.Close()
+
+	err = runSQLQuery(ctx, db)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("Example finished")
+}
+```
+
+This `main` function is pretty straightforward. It initializes a connection with
+the `otel-collector` service, which will be used by the tracer provider and the
+meter provider. Then, it configures the tracer provider and meter provider with
+the `connection` and a shutdown method, which ensures the telemetry data can be
+pushed to the `otel-collector` service correctly before exiting the application.
+After finishing setting up the OpenTelemetry client, it invokes the `connectDB`
+method to use the otelsql library to interact with the MySQL database. Let's
+look at the details here.
+
+```go
+func connectDB() *sql.DB {
+	// Connect to database
+	db, err := otelsql.Open("mysql", mysqlDSN, otelsql.WithAttributes(
+		semconv.DBSystemMySQL,
+	))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Register DB stats to meter
+	err = otelsql.RegisterDBStatsMetrics(db, otelsql.WithAttributes(
+		semconv.DBSystemMySQL,
+	))
+	if err != nil {
+		log.Fatal(err)
+	}
+	return db
+}
+```
+
+Instead of using [`sql.Open`](https://pkg.go.dev/database/sql#Open) method that
+Go offers, we use
+[`otelsql.Open`](https://pkg.go.dev/github.com/XSAM/otelsql#Open) to making an
+[`sql.DB`](https://pkg.go.dev/database/sql#DB) instance. The `sql.DB` instance
+returned by `otelsql.Open` is a wrapper that transfers and instruments all DB
+operations to the underlying `sql.DB` instance (created by `sql.Open`). When
+users send SQL queries with this wrapper, the `otelsql` can know the queries and
+use the OpenTelemetry client to instrument telemetry data.
```

Comment:
```suggestion
Instead of using the [`sql.Open`](https://pkg.go.dev/database/sql#Open) method
that Go provides, we use
[`otelsql.Open`](https://pkg.go.dev/github.com/XSAM/otelsql#Open) to create an
[`sql.DB`](https://pkg.go.dev/database/sql#DB) instance. The `sql.DB` instance
returned by `otelsql.Open` is a wrapper that transfers and instruments all DB
operations to the underlying `sql.DB` instance (created by `sql.Open`). When
users send SQL queries with this wrapper, `otelsql` can see the queries and
use the OpenTelemetry client to generate telemetry.
```